sdk/keyvault/keyvault-admin/src/accessControlClient.ts

```diff
@@ -453,22 +453,19 @@ export class KeyVaultAccessControlClient {
    */
   public upsertRoleDefinition(
     roleScope: KeyVaultRoleScope,
-    name: string,
-    permissions: KeyVaultPermission[],
-    description?: string,
     options: UpsertRoleDefinitionOptions = {}
   ): Promise<KeyVaultRoleDefinition> {
     return withTrace("upsertRoleDefinition", options, async (updatedOptions) => {
       const response = await this.client.roleDefinitions.createOrUpdate(
         this.vaultUrl,
         roleScope,
-        name,
+        options.roleDefinitionName || v4uuid(),
         {
           properties: {
-            description,
-            permissions,
+            description: options.description,
+            permissions: options.permissions,
             assignableScopes: [roleScope],
-            roleName: name,
+            roleName: options.roleName || `role_definition_name`,
```

Comment:
We decided not to default this - let the service do it if they want a value here.